core/drivers/versal_trng.c

```diff
@@ -963,6 +991,10 @@ static TEE_Result trng_kat_test(struct versal_trng *trng)
 	if (!trng)
 		return TEE_ERROR_GENERIC;
 
+	/* Self-tests are performed in HW */
+	if (trng->cfg.version == TRNG_V2)
+		return TEE_SUCCESS;
```

Comment:
All the self tests are still needed.

See https://github.com/Xilinx/embeddedsw/blob/xilinx_v2024.2/XilinxProcessorIPLib/drivers/trngpsx/src/xtrngpsx_tests.c#L211

Reply:
Linkage into the PKI: https://github.com/Xilinx/embeddedsw/blob/xilinx_v2024.2/lib/sw_services/xilpki/src/xilpki.c#L602